code/modules/reagents/chemistry/reagents/drug_reagents.dm

```diff
@@ -491,3 +491,127 @@
 	if(prob(15))
 		M.adjustToxLoss(2, 0)
 	..()
+
+/datum/reagent/drug/maint/on_mob_life(mob/living/carbon/M)
+	. = ..()
+
+
+/datum/reagent/drug/maint/addiction_act_stage1(mob/living/M)
+	. = ..()
+	M.Jitter(1)
+
+/datum/reagent/drug/maint/addiction_act_stage2(mob/living/M)
+	. = ..()
+	M.Jitter(2)
+	if(prob(15))
+		M.emote(pick("twitch","drool"))
+
+/datum/reagent/drug/maint/addiction_act_stage3(mob/living/M)
+	. = ..()
+	M.Jitter(1)
+	M.adjustToxLoss(2)
+	if(prob(5))
+		M.drop_all_held_items()
+	if(prob(15))
+		M.emote(pick("twitch","drool"))
+
+/datum/reagent/drug/maint/addiction_act_stage4(mob/living/M)
+	. = ..()
+	M.Jitter(2)
+	M.adjustToxLoss(3)
+	if(prob(10))
+		M.drop_all_held_items()
+	if(prob(30))
+		M.emote(pick("twitch","drool"))
+
+/datum/reagent/drug/maint/powder
+	name = "Maintanance Powder"
+	description = "An unknown powder that you most likely gotten from an assistant, a bored chemist... or cooked yourself. It is a refined form of tar that enhances your mental ability, making you learn stuff a lot faster."
+	reagent_state = SOLID
+	color = "#ffffff"
+	metabolization_rate = 0.5 * REAGENTS_METABOLISM
+	overdose_threshold = 15
+	addiction_threshold = 5
+
+/datum/reagent/drug/maint/powder/on_mob_metabolize(mob/living/L)
+	. = ..()
+	if(!iscarbon(L) || !L.mind)
+		return
+	var/mob/living/carbon/carbies = L
+	carbies.adjustOrganLoss(ORGAN_SLOT_BRAIN,0.1)
+	var/datum/mind/living_mind = carbies.mind
+	// 5x if you want to OD, you can potentially go higher, but good luck managing the brain damage.
+	living_mind.experience_modifier = round(volume/3,0.1) + 1
+
+/datum/reagent/drug/maint/powder/on_mob_end_metabolize(mob/living/M)
+	. = ..()
+	if(!M.mind)
+		return
+	var/datum/mind/living_mind = M.mind
+	living_mind.experience_modifier = initial(living_mind.experience_modifier)
+
+/datum/reagent/drug/maint/powder/overdose_process(mob/living/M)
+	. = ..()
+	if(!ishuman(M))
+		return
+	var/mob/living/carbon/human/humie = M
+	humie.adjustOrganLoss(ORGAN_SLOT_BRAIN,3)
+
+/datum/reagent/drug/maint/sludge
+	name = "Maintanance Sludge"
+	description = "An unknown sludge that you most likely gotten from an assistant, a bored chemist... or cooked yourself. Half refined, it fills your body with itself, making it more resistant to wound, but causes toxins to accumulate."
+	reagent_state = LIQUID
+	color = "#203d2c"
+	overdose_threshold = 25
+	addiction_threshold = 10
+
+/datum/reagent/drug/maint/sludge/on_mob_add(mob/living/L)
+	. = ..()
+	ADD_TRAIT(L,TRAIT_HARDLIMBDISABLE,type)
+
+/datum/reagent/drug/maint/sludge/on_mob_metabolize(mob/living/L)
```

Comment:
wait a minute, hold up

why is this in on_mob_metabolize() instead of on_mob_life()?

Reply:
and the addition of the hardlimbdisable trait is still in on_mob_add() instead of on_mob_metabolize()

Reply:
eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee